connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -468,13 +503,17 @@ public String workerId() {
     public void setTargetState(String connName, TargetState state) {
         log.info("Setting connector {} state to {}", connName, state);
 
-        WorkerConnector connector = connectors.get(connName);
-        if (connector != null)
-            connector.transitionTo(state);
-
-        for (Map.Entry<ConnectorTaskId, WorkerTask> taskEntry : tasks.entrySet()) {
-            if (taskEntry.getKey().connector().equals(connName))
-                taskEntry.getValue().transitionTo(state);
+        WorkerConnector workerConnector = connectors.get(connName);
+        // TODO: Make sure that tasks can't be around while a connector object is null
```

Comment:
Yeah, I thought so. 

Probably I'll add a getter to the `WorkerTask` to return the task's loader (which I inject in construction anyways).